hadoop-yarn-project/hadoop-yarn/bin/FederationStateStore/MySQL/FederationStateStoreStoredProcs.sql

```diff
@@ -122,10 +122,26 @@ BEGIN
    WHERE applicationId = applicationID_IN;
 END //
 
-CREATE PROCEDURE sp_getApplicationsHomeSubCluster()
-BEGIN
-   SELECT applicationId, homeSubCluster
-   FROM applicationsHomeSubCluster;
+CREATE PROCEDURE sp_getApplicationsHomeSubCluster(IN limit_IN int, IN homeSubCluster_IN varchar(256))
+BEGIN
+   SELECT
+       applicationId,
+       homeSubCluster,
+       createTime
+   FROM
+    (SELECT
+       applicationId,
+       homeSubCluster,
+       createTime,
+       @app_rank := IF(@home_sc = homeSubCluster, @app_rank + 1, 1) AS app_rank,
+       @home_sc := homeSubCluster
+  	 FROM applicationshomesubcluster
+  	 ORDER BY createTime DESC
+    ) ranked
+   WHERE app_rank <= limit_IN
+     AND (CASE WHEN t.homeSubCluster_IN IS NULL THEN 1 = 1
```

Comment:
Can we make this a regular boolean condition with no CASE?

Reply:
For MySQL, I also tried to refactor this SQL using a simpler way of writing.